kinder/ci/kinder-presubmit-tests.md

```diff
@@ -0,0 +1,59 @@
+# Kinder presubmit tests
+
+## Overview
+
+Kinder has a set of presubmit tests, designed in order to verify that:
+
+- Any PR sent to `kubernetes/kubeadm` repository with impact to the `kinder` codebase satisfies
+  all the expected code checks
+- Any PR sent to `kubernetes/kubeadm` repository with impact to the `kinder` codebase does not
+  compromise `kinder` features
+
+Kinder presubmit tests configuration can be found at <https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/kubeadm/kubeadm-presubmits.yaml>
+
+## Code Checks
+
+`pull-kubeadm-kinder-verify` is presubmits job that runs a set of code checks:
+
+- verify-whitespace
+- verify-spelling
+- verify-gofmt
+- verify-golint
+- verify-govet
+- verify-workflows
+- verify-deps
+- verify-gotest
+- verify-build
+
+All checks are implemented under the `hack` folder and the entry point is `hack/verify-all.sh`
+
+## Kinder feature checks
+
+`pull-kubeadm-kinder-init-versions` job checks that kinder can create cluster for the supported init versions:
+
+- current -3/minor (e.g. v1.11)
+- current -2/minor (e.g. v1.12)
+- current -1/minor (e.g. v1.13)
+- current          (e.g. v1.14)
+- master           (e.g. v1.15)
```

Comment:
can we get away by only running the `master` one?
having runs on all the versions can delay PR merging quite a bit.
- we are already going to have coverage for these in periodics
- we can debug locally.

Reply:
i'm not sure how fast it would be in CI, though.
with my internet it takes around 15 minutes to pull init + upgrade artifacts for the upgrade-stable-master job + a few more minutes to setup the cluster.

Reply:
something else we can do is have everything in the job, but only run in manually - e.g. `/test pull-kubeadm-kinder....`

Reply:
I'm open to any option, but on CI it takes
- 1.5 minutes average for pulling 2 kubernetes version (here we need only 1)
- 2 minutes average for init
Overall It should take less than 20 minutes

Reply:
ok let's try `always_run` as a start, but possibly amend if it ends up being too slow.